test/extensions/access_loggers/stats/integration_test.cc

```diff
@@ -0,0 +1,121 @@
+                      name: formatcounter
+                    value_format: '%RESPONSE_CODE%'
+                histograms:
+                  - stat:
+                      name: testhistogram
+                      tags:
+                        - name: tag
+                          value_format: '%REQUEST_HEADER(tag-value)%'
+                    value_format: '%REQUEST_HEADER(histogram-value)%'
+
+)EOF";
+
+  init(config_yaml);
+
+  Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "GET"},       {":authority", "envoyproxy.io"}, {":path", "/test/long/url"},
+      {":scheme", "http"},      {"tag-value", "mytagvalue"},     {"counter-value", "7"},
+      {"histogram-value", "2"},
+  };
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_EQ(response->headers().getStatusValue(), "200");
+
+  test_server_->waitForCounterEq(
+      "test_stat_prefix.fixedcounter.fixed_tag.fixed_value.dynamic_tag.mytagvalue_HTTP/1.1", 42);
+  test_server_->waitForCounterEq("test_stat_prefix.formatcounter", 200);
+  test_server_->waitUntilHistogramHasSamples("test_stat_prefix.testhistogram.tag.mytagvalue");
+
+  auto histogram = test_server_->histogram("test_stat_prefix.testhistogram.tag.mytagvalue");
+  EXPECT_EQ(1, TestUtility::readSampleCount(test_server_->server().dispatcher(), *histogram));
+  EXPECT_EQ(2, static_cast<uint32_t>(
+                   TestUtility::readSampleSum(test_server_->server().dispatcher(), *histogram)));
+}
+
+TEST_P(StatsAccessLogIntegrationTest, PercentHistogram) {
+  const std::string config_yaml = R"EOF(
+              name: envoy.access_loggers.stats
+              typed_config:
+                "@type": type.googleapis.com/envoy.extensions.access_loggers.stats.v3.Config
+                stat_prefix: test_stat_prefix
+                histograms:
+                  - stat:
+                      name: testhistogram
+                    unit: Percent
+                    value_format: '%REQUEST_HEADER(histogram-value)%'
+
+)EOF";
+
+  init(config_yaml);
+
+  Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "GET"},  {":authority", "envoyproxy.io"}, {":path", "/test/long/url"},
+      {":scheme", "http"}, {"histogram-value", "0.1"},
+  };
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_EQ(response->headers().getStatusValue(), "200");
+
+  test_server_->waitUntilHistogramHasSamples("test_stat_prefix.testhistogram");
+
+  auto histogram = test_server_->histogram("test_stat_prefix.testhistogram");
+  EXPECT_EQ(1, TestUtility::readSampleCount(test_server_->server().dispatcher(), *histogram));
+
+  double p100 = histogram->cumulativeStatistics().computedQuantiles().back();
+  EXPECT_NEAR(0.1, p100, 0.05);
+}
```

Comment:
Do you think we can exercise `emitLog` concurrently? Maybe spawn two workers and send ten requests? That would catch any unguarded data access problems.